src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -1,462 +1,445 @@
-    {
-        var paths = IOInputs.GetNonSignificantTrailingWhiteSpace();
-        foreach (var path in paths)
+        [Fact]
+        public void EmptyAsPath_ThrowsArgumentException()
         {
-            Assert.Throws<ArgumentException>(() =>          // BUG 995784: Not setting the parameter name
-            {
-                Directory.CreateDirectory(path);
-            });
+            Assert.Throws<ArgumentException>(() => Create(string.Empty));
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_PathWithInvalidCharactersAsPath_ThrowsArgumentException()
-    {
-        var paths = IOInputs.GetPathsWithInvalidCharacters();
-        foreach (var path in paths)
+        [Fact]
+        public void PathWithInvalidCharactersAsPath_ThrowsArgumentException()
         {
-            Assert.Throws<ArgumentException>(() =>          // BUG 995784: Not setting the parameter name
+            var paths = IOInputs.GetPathsWithInvalidCharacters();
+            Assert.All(paths, (path) =>
             {
-                Directory.CreateDirectory(path);
+                Assert.Throws<ArgumentException>(() => Create(path));
             });
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // alternate data streams
-    public static void CreateDirectory_PathWithAlternativeDataStreams_ThrowsNotSupportedException()
-    {
-        var paths = IOInputs.GetPathsWithAlternativeDataStreams();
-        foreach (var path in paths)
+        [Fact]
+        public void PathAlreadyExistsAsFile()
         {
-            Assert.Throws<NotSupportedException>(() =>
-            {
-                Directory.CreateDirectory(path);
-            });
+            string path = GetTestFilePath();
+            File.Create(path).Dispose();
+
+            Assert.Throws<IOException>(() => Create(path));
+            Assert.Throws<IOException>(() => Create(IOServices.AddTrailingSlashIfNeeded(path)));
+            Assert.Throws<IOException>(() => Create(IOServices.RemoveTrailingSlash(path)));
         }
-    }
 
-    [Fact]
-    [OuterLoop]
-    [PlatformSpecific(PlatformID.Windows)] // device name prefixes
-    public static void CreateDirectory_PathWithReservedDeviceNameAsPath_ThrowsDirectoryNotFoundException()
-    {   // Throws DirectoryNotFoundException, when the behavior really should be an invalid path
-        var paths = IOInputs.GetPathsWithReservedDeviceNames();
-        foreach (var path in paths)
+        [Theory]
+        [InlineData(FileAttributes.Hidden)]
+        [InlineData(FileAttributes.ReadOnly)]
+        [InlineData(FileAttributes.Normal)]
+        public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
         {
-            Assert.Throws<DirectoryNotFoundException>(() =>
-            {
-                Directory.CreateDirectory(path);
-            });
+            DirectoryInfo testDir = Create(GetTestFilePath());
+            testDir.Attributes = attributes;
```

Comment:
The original tests were split into four versions of "PathAlreadyExists": File, Directory, HiddenDirectory, ReadOnlyDirectory. I combined the last three tests into one. The tests should function the same otherwise, where the basic structure is:

Create file/directory/hiddendirectory/readonlydirectory
Create new directory
Check that the new directory is the same as the old directory, or catch an exception if the old path referred to a file instead.

Attribute setting is assumed to work correctly for the purposes of this test. You are correct that it is meaning to test preexisting files/directories.